### PR TITLE
[TwigComponents] Fixing inconsistency with how {% component rendered vs {{ component()

### LINKED ISSUE #1205: Unable to find custom template for UX components with <twig:component> syntax
### UX Twig Component version

2.12

### Description

When i define a custom twig template of a component, it's ok where I called the component with the classic syntax but it's not possible to find the template with the new <twig:component> syntax.

### How to reproduce

```bash
composer req symfony/ux-twig-component
```
(Version v2.12.0)

1- Configure a new folder for Twig
```yaml
# config/packages/twig.yaml
twig:
    paths:
        - '%kernel.project_dir%/templates'
        - '%kernel.project_dir%/components'
```
2- Configure PHP classes autoload (❗dump-autoload)
```json
    "autoload": {
        "psr-4": {
            "App\\": "src/",
            "Component\\": "components/"
        }
    },
```
and
```yaml
services:

    Component\:
        resource: '../components/'
```
3- Create a folder `components/Button`
4- Create a PHP file
```php
<?php
namespace Component\Button;

use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;

#[AsTwigComponent('button', template: 'button.twig')]
class ButtonComponent { }
```
5- Create a Twig file `components/button.twig`
```twig
<button>Wixiweb</button>
```

---

You can use this button like this : 
`{{ component('button') }}`

But you cannot use this button like this : 
`<twig:button></twig:button>`

> Unable to find template "button.twig" (looked into: /templates, /components, /templates) in [home/index.twig at line 13]

---
From : https://github.com/symfony/symfony/issues/52154

### PR DESCRIPTION
**Pull request by weaverryan**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| Issues        | Fix #1205 (?) and maybe others?
| License       | MIT

tl;dr `{% component` was rendered slightly differently than `{{ component`. Additionally, for `{% component`, the template was determined too early, which didn't allow for the template to be overridden via an event.

**Longer Explanation**

This cleans up `ComponentTokenParser`, removing some unnecessary parts. We now always set the "embedded" component parent to a magic `__parent__` variable. This allows us to delay the determination of the component template until runtime, after the `PreRenderEvent` has been called.

TODO:

* [ ] Bring in tests from #1236 and #1237 and check other tests
* [ ] Check #1205 to verify this fixes
* [ ] Check debug:twig-component command for any changes needed

Cheers!

### CODE REVIEW

**Comment by smnandre** [comment]
Just tried, both "TwigComponent should pass" tests are OK

**Comment by smnandre** [comment]
DebugCommand seems OK after a couple of tests

**Comment by weaverryan** [comment]
@smnandre would you mind rebasing your 2 PR's? I'd like to verify they pass - then we can merge those in

**Comment by smnandre** [comment]
> @smnandre would you mind rebasing your 2 PR's? I'd like to verify they pass - then we can merge those in

Will do in 2/3 hours
